intranet/apps/announcements/tests.py

```diff
@@ -94,3 +105,371 @@ def test_announcement_request(self):
             ).count(),
             1,
         )
+
+    def test_request_announcement_success_view(self):
+        self.login()
+        response = self.client.get(reverse("request_announcement_success"))
+        self.assertEqual(200, response.status_code)
+
+    def test_request_announcement_success_self_view(self):
+        self.login()
+        response = self.client.get(reverse("request_announcement_success_self"))
+        self.assertEqual(200, response.status_code)
+        self.assertTrue(response.context["self"])
+
+    def test_approve_announcement_view(self):
+        teacher = get_user_model().objects.get_or_create(username="awilliam1", user_type="teacher", first_name="a", last_name="william")[0]
+        user = self.login()
+        request = AnnouncementRequest.objects.create(title="test", content="hello", author="Sysadmins", user=user)
+
+        self.login("awilliam1")
+
+        # The user is not in the "requested_teachers" list, so this should 302 to the index page
+        response = self.client.get(reverse("approve_announcement", kwargs={"req_id": request.id}), follow=True)
+        self.assertEqual(200, response.status_code)
```

Comment:
I believe you could better test that it really `302`s by first making the request without `follow=True`. So:
```
 response = self.client.get(reverse("approve_announcement", kwargs={"req_id": request.id}))
 self.assertEqual(302, response.status_code)
```
and then what you have here.

Reply:
The check for `response.context["messages"]` (the next line) does not work without `follow=True` and asserting a 200 response, I believe. But I'll try to see.

Reply:
No, that line won't work without `follow=True`. I'm suggesting leaving what you have and add the two lines I wrote above line 128.

Reply:
Oh... okay. Makes sense.